synapse/http/server.py

```diff
@@ -706,15 +707,35 @@ def respond_with_json_bytes(
     if send_cors:
         set_cors_headers(request)
 
-    # note that this is zero-copy (the bytesio shares a copy-on-write buffer with
-    # the original `bytes`).
-    bytes_io = BytesIO(json_bytes)
-
-    producer = NoRangeStaticProducer(request, bytes_io)
-    producer.start()
+    _write_json_bytes_to_request(request, json_bytes)
```

Comment:
From https://github.com/matrix-org/synapse/pull/10844#discussion_r713039021

> why is this changing in this PR? It seems unrelated to how we encode JSON responses?

True, I can leave as is, but it just felt cleaner to have both `respond_with_json` and `respond_with_json_bytes` use the same producer. Leaving as is does mean a) returning large responses can lead to timeouts, and b) anything that writes responses via this method in the replication unit tests will fail (due to `Request` converting pull producers to push producers via the default reactor, which is never run. I am sorely tempted to ban all pull producers because I lost like half a day due to that, but I digress).

Reply:
:+1: